src/AutoRest.CSharp/Common/Output/Models/Requests/RequestHeader.cs

```diff
@@ -32,9 +32,20 @@ internal class RequestHeader
             RepeatabilityRequestId,
             RepeatabilityFirstSent,
         };
-
         public static bool IsRepeatabilityRequestHeader(string headerName) => RepeatabilityRequestHeaders.Contains(headerName);
 
+        public static HashSet<string> ClientRequestIdHeaders = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "client-request-id",
+        };
+
+        public static HashSet<string> ReturnClientRequestIdResponseHeaders = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "return-client-request-id",
+        };
+
+        public static bool IsClientRequestIdHeader(string headerName) => ClientRequestIdHeaders.Contains(headerName) || ReturnClientRequestIdResponseHeaders.Contains(headerName);
```

Comment:
If using `.Contains()`, then we can just use one `HashSet` to host both `client-request-id` and `return-client-request-id`.

Reply:
we handle `client-request-id` and `return-client-request-id` differently. (the different constant expression), so I separate them.

Reply:
You can define them as constant, and still put them into the same HashSet. See https://github.com/Azure/autorest.csharp/blob/8fd5781ccb57e422aaccd918270be3ffc6bd0d6c/src/AutoRest.CSharp/Common/Output/Models/Requests/RequestHeader.cs#L27-L34

That can also simplify the logic in `Parameter.cs` a little bit.

Reply:
@archerzz I prefer to separate clientRequestId and returnClientRequstID into different set. They are different. And I define ClinetRequestID a set (not a string available) because it may be customized to another head name instead of `client-request-id`. And in the future, we will remove ClientRequestIDPolicy from pipeline, and then `x-ms-client-request-id` will included and handle the same way.